src/test/java/jenkins/plugins/logstash/persistence/LogzioDaoTest.java

```diff
@@ -0,0 +1,190 @@
+
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import static org.mockito.Mockito.*;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+@RunWith(PowerMockRunner.class)
+@PowerMockIgnore({"javax.crypto.*"})
+@PrepareForTest(LogstashConfiguration.class)
+public class LogzioDaoTest {
+
+    private static final String data = "{\"a\":{\"b\":1,\"c\":2,\"d\":[false, true]},\"e\":\"f\",\"g\":2.3}";
+    private static final String flat_data = "\"a_b\":1,\"a_c\":2,\"a_d[0]\":false,\"a_d[1]\":true,\"e\":\"f\",\"g\":2.3";
+    private static final String EMPTY_STRING_WITH_DATA = "{\"@buildTimestamp\":\"2000-01-01\"," + flat_data + ",\"message\":[],\"source\":\"jenkins\",\"source_host\":\"http://localhost:8080/jenkins\",\"@version\":1}";
+    private static final String ONE_LINE_STRING_WITH_DATA = "{\"@buildTimestamp\":\"2000-01-01\"," + flat_data + ",\"message\":[\"LINE 1\"],\"source\":\"jenkins\",\"source_host\":\"http://localhost:8080/jenkins\",\"@version\":1}";
+    private static final String TWO_LINE_STRING_WITH_DATA = "{\"@buildTimestamp\":\"2000-01-01\"," + flat_data + ",\"message\":[\"LINE 1\", \"LINE 2\"],\"source\":\"jenkins\",\"source_host\":\"http://localhost:8080/jenkins\",\"@version\":1}";
+    private static final String EMPTY_STRING_NO_DATA = "{\"@buildTimestamp\":\"2000-01-01\",\"message\":[],\"source\":\"jenkins\",\"source_host\":\"http://localhost:8080/jenkins\",\"@version\":1}";
+    private static final String ONE_LINE_STRING_NO_DATA = "{\"@buildTimestamp\":\"2000-01-01\",\"message\":[\"LINE 1\"],\"source\":\"jenkins\",\"source_host\":\"http://localhost:8080/jenkins\",\"@version\":1}";
+    private static final String TWO_LINE_STRING_NO_DATA = "{\"@buildTimestamp\":\"2000-01-01\",\"message\":[\"LINE 1\", \"LINE 2\"],\"source\":\"jenkins\",\"source_host\":\"http://localhost:8080/jenkins\",\"@version\":1}";
+    private LogzioDao dao;
+
+    @Captor private ArgumentCaptor<JsonObject> sendArgument = ArgumentCaptor.forClass(JsonObject.class);
+
+    @Mock private LogzioSender logzioSender;
+    @Mock private BuildData mockBuildData;
+    @Mock private LogstashConfiguration logstashConfiguration;
+
+    private LogzioDao createDao(String host, String key) throws IllegalArgumentException {
+        return new LogzioDao(logzioSender, host, key);
+    }
+
+    @Before
+    public void before() throws IllegalArgumentException {
+        PowerMockito.mockStatic(LogstashConfiguration.class);
+        when(LogstashConfiguration.getInstance()).thenReturn(logstashConfiguration);
+        when(logstashConfiguration.getDateFormatter()).thenCallRealMethod();
```

Comment:
it seems to me you are mocking the object only to tell it to `thenCallRealMethod` -- seems unnecessary

Reply:
I tried without and it's not working. I think because I'm mocking the whole class

Reply:
You could use a JenkinsRule then you have a Jenkins instance